app/services/doc_auth/lexis_nexis/responses/true_id_response.rb

```diff
@@ -37,7 +37,8 @@ def initialize(http_response, config, liveness_checking_enabled = false)
         #    document type
         #    bar code attention
         def successful_result?
-          (all_passed? || attention_with_barcode?) && id_type_supported?
+          doc_auth_success? &&
+            (@liveness_checking_enabled ? selfie_passed? : true)
```

Comment:
Hmm I'm sorry to wave back on my approval @amirbey, after reviewing a comment by @eileen-nava on the previous iteration of this code, I'm not sure this change would cover the `attention_with_barcode?` case (tagged you in the conversation on the doc).

I did find when changing the logic to `all_passed?` to account for selfie, that this `successful_result?` didn't have tests. I [added them for the selfie](https://github.com/18F/identity-idp/blob/main/spec/services/doc_auth/lexis_nexis/responses/true_id_response_spec.rb#L697), but I wonder if now with all these questions on the logic, if we really need to add tests for the other cases (ie, no selfie but attention with barcode, etc).

I wasn't sure if I should change my review to "request changes", but I'm not sure if that should / could be a separate ticket.